### PR TITLE
Host Release prep 4.36.0

### PR DESCRIPTION
**Pull request by v-imohammad**
### Updated the Common.prop with minor version and updated the release notes well

resolves #issue_for_this_pr

### Pull request checklist

**IMPORTANT**: Currently, changes must be backported to the `in-proc` branch to be included in Core Tools and non-Flex deployments.

* [ ] Backporting to the `in-proc` branch is not required
    * Otherwise: Link to backporting PR
* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] My changes **do not** require diagnostic events changes
    * Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: build/common.props [hunk 1]
```diff
@@ -3,7 +3,7 @@
   <PropertyGroup>
     <LangVersion>latest</LangVersion>
     <MajorVersion>4</MajorVersion>
-    <MinorVersion>$(MinorVersionPrefix)35</MinorVersion>
+    <MinorVersion>$(MinorVersionPrefix)36</MinorVersion>
     <PatchVersion>4</PatchVersion>
```
Comment (kshyju) [inline build/common.props]: Shouldn't we set the patch version to 0?

Comment (v-imohammad) [reply build/common.props]: @kshyju  I overlooked the patch version, so thank you for pointing it out. Your keen observation is greatly appreciated. To address this, I've added another commit. Thank you.

File: release_notes.md [hunk 2]
```diff
@@ -8,24 +8,8 @@
   - Fix thread context classloader for middleware chain
 - Update Java Worker Version to [2.15.0](https://github.com/Azure/azure-functions-java-worker/releases/tag/2.15.0):
   - Update grpc-protobuf to 1.64.0 and application insights agent version to 3.5.2
-- Marking SyncTriggers [RequiresRunningHost]. (#10233)
```
Comment (fabiocav) [inline release_notes.md:11]: Can you clarify why some of the release notes are being removed? Was this not reset during the last release?

Comment (v-imohammad) [reply release_notes.md:11]: @fabiocav  This commit mentioned in the release notes for this PR was already included in version 4.35.x, which is why it has been excluded from the release notes for the upcoming version 4.36.x.

File: release_notes.md [hunk 3]
```diff
@@ -8,24 +8,8 @@
...
-- Defaulting SwtAuthenticationEnabled to False (#10195)
-- Adding runtime site name to valid JWT audiences (slot scenarios) (#10185)
 - Update Python Worker Version to [4.30.3](https://github.com/Azure/azure-functions-python-worker/releases/tag/4.30.3)
-- Language worker channels will not be started during placeholder mode if we are in-process (#10161)
 - Ordered invocations are now the default (#10201)
```
Comment (kshyju) [inline release_notes.md]: "Ordered invocations are now the default" - Was this change already included in version 4.35.4?

Comment (v-imohammad) [reply release_notes.md]: @kshyju Yeap this is part of the release notes 4.35.4
https://github.com/Azure/azure-functions-host/releases/tag/v4.35.4

Comment (kshyju) [reply release_notes.md]: Shouldn't that be removed in that case , for 4.36.0 ?

Comment (v-imohammad) [reply release_notes.md]: @kshyju  we removed this from release notes

File: release_notes.md [hunk 4]
```diff
@@ -8,29 +8,15 @@
...
   - `Azure.Security.KeyVault.Secrets` updated to 4.6.0
   - `System.Format.Asn1` updated to 6.0.1
 - Update PowerShell 7.2 worker to [4.0.4020](https://github.com/Azure/azure-functions-powershell-worker/releases/tag/v4.0.4020)
 - Update PowerShell 7.4 worker to [4.0.4021](https://github.com/Azure/azure-functions-powershell-worker/releases/tag/v4.0.4021)
-- Trim FunctionsNetHost artifacts (#10364)
 - Resolved thread safety issue in the `GrpcWorkerChannel.LoadResponse` method. (#10352)
 - Worker termination path updated with sanitized logging (#10397)
+- [in-proc] Avoid redundant DiagnosticEvents error message (#10396)
+- Fixing thread safety issue in the GrpcWorkerChannel.LoadResponse method (#10363) (#10400)
+- Updating DotNetIsolatedNativeHost package to 1.0.11 (#10380)
```
Comment (kshyju) [inline release_notes.md]: Is this not already part of [4.35.4](https://github.com/Azure/azure-functions-host/releases/tag/v4.35.4)?

Comment (v-imohammad) [reply release_notes.md]: @kshyju These were not included in 4.35.4 release notes

Comment (kshyju) [reply release_notes.md]: Sorry! What do you mean they were not included? It is listed in the release notes I linked above. You can also see the commit here in 4.35.4 tag - https://github.com/Azure/azure-functions-host/blob/v4.35.4/src/WebJobs.Script/WebJobs.Script.csproj#L56

Comment (v-imohammad) [reply release_notes.md]: @kshyju I appreciate your guidance on this matter. However, as mentioned, we typically rely on release notes for information. Unfortunately, the details from version 4.35.4 were not included in the release notes, even though they were present in the code.

File: release_notes.md (hunk 2 above)
Comment (fabiocav) [reply release_notes.md:11]: Can you clarify where you're pulling the release notes from? I wouldn't expect this state as part of release prep.

Comment (v-imohammad) [reply release_notes.md:11]: @fabiocav  Typically, we don't include the addition process, but this time we've meticulously cross-checked the list of commits made after the 4.35.0 release. We then compared it with the release notes of 4.35.4 and filtered it from the existing release notes.
https://github.com/Azure/azure-functions-host/releases
image

**Review by gavin-aguiar — APPROVED** [review]

**Review by kamperiadis — APPROVED** [review]

File: release_notes.md (hunk 2 above)
Comment (fabiocav) [reply release_notes.md:11]: Updated the release notes.

**Review by fabiocav — APPROVED** [review]

**Pull request merged**